code/modules/hivemind/machines.dm

```diff
@@ -0,0 +1,486 @@
+
+
+//there we binding or un-binding hive with wire
+//in this way, when our node will be destroyed, wireweeds will die too
+/obj/structure/hivemind_machine/node/proc/add_wireweed(obj/effect/plant/hivemind/wireweed)
+	if(wireweed.master_node)
+		wireweed.master_node.remove_wireweed(wireweed)
+	wireweed.master_node = src
+	my_wireweeds.Add(wireweed)
+
+/obj/structure/hivemind_machine/node/proc/remove_wireweed(obj/effect/plant/hivemind/wireweed)
+	my_wireweeds.Remove(wireweed)
+	wireweed.master_node = null
+
+//there we check for other nodes
+//if no any other hives will be found, game over
+/obj/structure/hivemind_machine/node/proc/check_for_other()
+	if(hive_mind_ai)
+		if(!hive_mind_ai.hives.len)
+			hive_mind_ai.die()
+
+
+//TURRET
+//shooting the target with toxic goo
+/obj/structure/hivemind_machine/turret
+	name = "shooter"
+	desc = "Strange thing with some kind of tube."
+	max_health = 140
+	icon_state = "turret"
+	cooldown_time = 5 SECONDS
+	var/proj_type = /obj/item/projectile/goo
+
+
+/obj/structure/hivemind_machine/turret/activate()
+	for(var/mob/living/target in view(src))
+		if(target.stat == CONSCIOUS && target.faction != "hive")
+			use_ability(target)
+			set_cooldown()
+			break
+
+
+/obj/structure/hivemind_machine/turret/use_ability(atom/target)
+	var/obj/item/projectile/proj = new proj_type(loc)
+	proj.launch(target)
+	playsound(src, 'sound/effects/blobattack.ogg', 70, 1)
+
+
+
+//MOB PRODUCER
+//spawns mobs from list
+/obj/structure/hivemind_machine/mob_spawner
+	name = "assembler"
+	desc = "Cylindrical machine with some lights and entry port. You can hear how something moves inside."
+	max_health = 120
+	icon_state = "spawner"
+	cooldown_time = 10 SECONDS
+	var/mob_to_spawn
+	var/mob_amount = 1
+
+/obj/structure/hivemind_machine/mob_spawner/Initialize()
+	..()
+	mob_to_spawn = pick(/mob/living/simple_animal/hostile/hivemind/stinger, /mob/living/simple_animal/hostile/hivemind/bomber)
+
+
+/obj/structure/hivemind_machine/mob_spawner/activate()
+	if(!mob_to_spawn || spawned_creatures.len >= mob_amount)
+		return
+	if(locate(/mob/living) in loc)
+		return
+
+	//here we upgrading our spawner and rise controled mob amount, based on EP
+	if(hive_mind_ai.evo_points > 100)
+		mob_amount = 2
+	else if(hive_mind_ai.evo_points > 300)
+		mob_amount = 3
+
+	for(var/mob/living/target in view(src))
+		if(target.stat != DEAD && target.faction != "hive")
+			use_ability()
+			set_cooldown()
+			break
+
+
+/obj/structure/hivemind_machine/mob_spawner/use_ability()
+	var/mob/living/simple_animal/hostile/hivemind/spawned_mob = new mob_to_spawn(loc)
+	spawned_creatures.Add(spawned_mob)
+	spawned_mob.master = src
+	flick("[icon_state]-anim", src)
+
+
+
+//MACHINE PREACHER
+//creepy radio talk, it's okay if they have no sense sometimes
+/obj/structure/hivemind_machine/babbler
+	name = "connector"
+	desc = "Column-like structure with lights. You can see how energy stream moves inside."
```

Comment:
"A column-like structure with lights. You can see streams of energy moving inside."